src/Microsoft.Identity.Client/Azure/TokenProvider.cs

```diff
@@ -0,0 +1,43 @@
+using Microsoft.Identity.Client.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Azure
+{
+    /// <summary>
+    /// ITokenProvider describes the interface for fetching an access token
+    /// </summary>
+    public interface ITokenProvider
```

Comment:
Should it be public?

Reply:
addressed in: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/924#discussion_r263524751